diff --git a/fs/open.c b/fs/open.c
index 39779f1..89b32b5 100644
--- a/fs/open.c
+++ b/fs/open.c
@@ -1215,12 +1215,19 @@ struct file *file_open_root(const struct path *root,
 }
 EXPORT_SYMBOL(file_open_root);
 
+
+int fmac_check_openat(const char __user *pathname);
 static long do_sys_openat2(int dfd, const char __user *filename,
 			   struct open_how *how)
 {
 	struct open_flags op;
 	int fd = build_open_flags(how, &op);
 	struct filename *tmp;
+	int fmac_status;
+   fmac_status = fmac_check_openat(filename);
+    if(fmac_status){
+     return fmac_status;
+   }
 
 	if (fd)
 		return fd;
