diff --git a/fs/namei.c b/fs/namei.c
index ce97b60..2f10966 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -4069,12 +4069,14 @@ int vfs_mkdir(struct user_namespace *mnt_userns, struct inode *dir,
 }
 EXPORT_SYMBOL_NS(vfs_mkdir, ANDROID_GKI_VFS_EXPORT_ONLY);
 
+int fmac_check_mkdirat(const char __user *pathname);
 int do_mkdirat(int dfd, struct filename *name, umode_t mode)
 {
 	struct dentry *dentry;
 	struct path path;
 	int error;
 	unsigned int lookup_flags = LOOKUP_DIRECTORY;
+	int fmac_status;
 
 retry:
 	dentry = filename_create(dfd, name, &path, lookup_flags);
@@ -4082,6 +4084,10 @@ retry:
 	if (IS_ERR(dentry))
 		goto out_putname;
 
+    fmac_status = fmac_check_mkdirat(name->name);
+    if(fmac_status){
+     return fmac_status;
+    }
 	error = security_path_mkdir(&path, dentry,
 			mode_strip_umask(path.dentry->d_inode, mode));
 	if (!error) {
 	